pkg/types/azure/machinepool.go

```diff
@@ -38,4 +44,18 @@ func (a *MachinePool) Set(required *MachinePool) {
 	if required.OSDisk.DiskSizeGB != 0 {
 		a.OSDisk.DiskSizeGB = required.OSDisk.DiskSizeGB
 	}
+
+	a.OSDisk.DiskType = validateDiskType(required.OSDisk.DiskType)
```

Comment:
If we were to error out if invalid value is given then I think this code should be in here since it's azure specific.
https://github.com/openshift/installer/blob/6f9a0b8721d3cadfca938e9f4abc9bba3ac292ac/pkg/types/azure/validation/machinepool.go#L14

Making the changes.